libs/pages/layout/src/lib/feature/layout/layout.tsx

```diff
@@ -50,13 +50,13 @@ export function Layout(props: LayoutProps) {
         if (result.length > 0 && !organizationIds.includes(organizationId)) {
           dispatch(fetchOrganizationById({ organizationId }))
             .unwrap()
-            .catch(() => navigate(ORGANIZATION_URL(result[0].id)))
+            .catch(() => redirect(ORGANIZATION_URL(result[0].id))) // using redirect instead of navigate because navigate was needed in useEffect deps and retriggering everytime we change location https://github.com/remix-run/react-router/discussions/8465#discussioncomment-4051081
         }
       })
       .catch((error) => console.error(error))
 
     dispatch(fetchUserSignUp())
-  }, [dispatch, organizationId, navigate])
+  }, [dispatch, organizationId])
```

Comment:
Missing redirect ?

Reply:
no, redirect is not declared inside our component, it's an import directly

Reply:
https://reactrouter.com/en/main/fetch/redirect